bindings/rust/bench/src/rustls.rs

```diff
@@ -82,23 +87,38 @@ impl TlsBenchHarness for RustlsHarness {
             ECGroup::X25519 => &X25519,
         };
 
-        let client_config = Arc::new(
-            ClientConfig::builder()
-                .with_cipher_suites(&[cipher_suite])
-                .with_kx_groups(&[kx_group])
-                .with_protocol_versions(&[&TLS13])?
-                .with_root_certificates(Self::get_root_cert_store()?)
-                .with_no_client_auth(),
-        );
-
-        let server_config = Arc::new(
-            ServerConfig::builder()
-                .with_cipher_suites(&[cipher_suite])
-                .with_kx_groups(&[kx_group])
-                .with_protocol_versions(&[&TLS13])?
-                .with_no_client_auth()
-                .with_single_cert(Self::get_cert_chain()?, Self::get_server_key()?)?,
-        );
+        let client_builder = ClientConfig::builder()
+            .with_cipher_suites(&[cipher_suite])
+            .with_kx_groups(&[kx_group])
+            .with_protocol_versions(&[&TLS13])?
+            .with_root_certificates(Self::get_root_cert_store()?);
+
+        let server_builder = ServerConfig::builder()
+            .with_cipher_suites(&[cipher_suite])
+            .with_kx_groups(&[kx_group])
+            .with_protocol_versions(&[&TLS13])?;
+
+        let (client_builder, server_builder) = match handshake_type {
+            HandshakeType::mTLS => (
+                client_builder.with_single_cert(
+                    Self::get_cert_chain(CLIENT_CERT_CHAIN_PATH)?,
+                    Self::get_key(CLIENT_KEY_PATH)?,
+                )?,
+                server_builder.with_client_cert_verifier(Arc::new(
+                    AllowAnyAuthenticatedClient::new(Self::get_root_cert_store()?),
+                )),
+            ),
+            HandshakeType::Full => (
+                client_builder.with_no_client_auth(),
+                server_builder.with_no_client_auth(),
+            ),
+        };
+
+        let client_config = Arc::new(client_builder);
+        let server_config = Arc::new(server_builder.with_single_cert(
```

Comment:
Why did you move the with_single_cert function call here? IMO it looks better if all the builder stuff is in the same place.

Reply:
Rustls requires the builder to be built in a specific order, so `with_single_cert()` for the server config has to happen after the mTLS match block.